### PR TITLE
Add test for sealed

### LINKED ISSUE #1534: bus error with `sealed!`
#### Input

```ruby
# typed: true

extend T::Sig

module Boolean
  sealed!
end

sig {params(x: Boolean).void}
def foo(x)
  case x
  when TrueClass
    T.reveal_type(x)
  else
    T.absurd(x)
  end
end
```

#### Observed output

bus error

#### Expected behavior

no bus error

### PR DESCRIPTION
**Pull request by jez**
### Motivation

Originally from https://github.com/sorbet/sorbet/issues/1534

I don't think the bug that was reported in the ticket happens anymore,
so I'm adding it as a test.

Closes #1534

### Test plan

See included automated tests.

### CODE REVIEW

**Review by elliottt — APPROVED** [review]

File: test/testdata/infer/fuzz_sealed.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# typed: true
+
+extend T::Sig
+
+module Boolean
+  sealed!
```
Comment (elliottt) [inline test/testdata/infer/fuzz_sealed.rb:6]:
```suggestion
  extend T::Helpers
  sealed!
```

Was this meant to raise an error?

Comment (jez) [reply test/testdata/infer/fuzz_sealed.rb:6]: I took this verbatim from @azdavis's report.

It's possible that missing the `extend T::Helpers` was precisely the problem?

Comment (jez) [inline test/testdata/infer/fuzz_sealed.rb:6]:
```suggestion
  sealed! # error: `sealed!` does not exist
```

Comment (elliottt) [reply test/testdata/infer/fuzz_sealed.rb]: Ah cool, thanks!

**Review by elliottt — APPROVED** [review]

File: test/testdata/infer/fuzz_sealed.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+module Boolean
+  sealed! # error: `sealed!` does not exist
+end
+
+sig {params(x: Boolean).void}
+def foo(x)
+  case x
+  when TrueClass
+    T.reveal_type(x)
```
Comment (jez) [inline test/testdata/infer/fuzz_sealed.rb:13]:
```suggestion
    T.reveal_type(x) # error: Revealed type: `T.all(Boolean, TrueClass)`
```

**Pull request merged**